esp-hal/src/gpio/mod.rs

```diff
@@ -987,6 +1007,27 @@ macro_rules! io_type {
     (Output, $gpionum:literal) => {
         impl $crate::gpio::OutputPin for GpioPin<$gpionum> {}
     };
+    (Analog, $gpionum:literal) => {
+        // FIXME: the implementation shouldn't be in the GPIO module
+        #[cfg(any(esp32c2, esp32c3, esp32c6, esp32h2))]
+        impl $crate::gpio::AnalogPin for GpioPin<$gpionum> {
+            /// Configures the pin for analog mode.
+            fn set_analog(&self, _: $crate::private::Internal) {
```

Comment:
I hope to see this method's implementation can be moved into the `AnalogPin` trait itself in future.